File: final-value-of-variable/solution2011.go
// https://leetcode.com/problems/final-value-of-variable-after-performing-operations/

package main

import "fmt"

// T: O(n), S: O(1)
func finalValueAfterOperations(operations []string) int {
	x := 0
	for _, op := range operations {
		switch op {
		case "X++", "++X":
			x++
		case "X--", "--X":
			x--
		}
	}
	return x
}

func main() {
	ops1 := []string{"--X", "X++", "X++"}
	ops2 := []string{"++X", "++X", "X++"}
	fmt.Println(finalValueAfterOperations(ops1)) // 1
	fmt.Println(finalValueAfterOperations(ops2)) // 3
}

File: final-value-of-variable/Solution2011.java
// https://leetcode.com/problems/final-value-of-variable-after-performing-operations/

class Solution2011 {

    // T: O(n), S: O(1)
    public int finalValueAfterOperations(String[] operations) {
        int x = 0;
        for (String op : operations) {
            switch (op) {
                case "X++":
                case "++X":
                    x++;
                    break;
                case "X--":
                case "--X":
                    x--;
                    break;
            }
        }
        return x;
    }

    public static void main(String[] args) {
        var sol = new Solution2011();
        var ops1 = new String[] { "--X", "X++", "X++" };
        var ops2 = new String[] { "++X", "++X", "X++" };
        System.out.println(sol.finalValueAfterOperations(ops1)); // 1
        System.out.println(sol.finalValueAfterOperations(ops2)); // 3
    }

}
